Building with Qt6.7.1 needs [1] and a part of [2]+[3]. [2] was intended
to fix building with Qt6.8 but turned out part of it was needed for 6.7.1
and this was adjusted in [3]. [2] does not apply cleanly so only taking
the bit needed given 6.8 can wait for a proper release.

[1] https://github.com/qt-creator/qt-creator/commit/041727fb2
[2] https://github.com/qt-creator/qt-creator/commit/6babc022d
[3] https://github.com/qt-creator/qt-creator/commit/fcf2d2064
--- a/src/libs/utils/fsengine/fsengine_impl.h
+++ b/src/libs/utils/fsengine/fsengine_impl.h
@@ -52,4 +52,7 @@
     uint ownerId(FileOwner) const override;
     QString owner(FileOwner) const override;
+    #if QT_VERSION >= QT_VERSION_CHECK(6, 7, 1)
+        using FileTime = QFile::FileTime;
+    #endif
     bool setFileTime(const QDateTime &newDate, FileTime time) override;
     QDateTime fileTime(FileTime time) const override;
--- a/src/plugins/languageclient/languageclientutils.cpp
+++ b/src/plugins/languageclient/languageclientutils.cpp
@@ -236,6 +236,6 @@
     Client *client = LanguageClientManager::clientForDocument(textEditor->textDocument());
 
-    ClientExtras *extras = widget->findChild<ClientExtras *>(clientExtrasName,
-                                                             Qt::FindDirectChildrenOnly);
+    ClientExtras *extras = dynamic_cast<ClientExtras *>(
+        widget->findChild<QObject *>(clientExtrasName, Qt::FindDirectChildrenOnly));
     if (!extras) {
         if (!client)
